src/APIBasedAuth.ts

```diff
@@ -133,6 +133,18 @@ class APIBasedAuth {
     return data;
   }
 
+  public async getLoginMethods(username: string) {
+    if (this.loginMethods?.[username]) {
+      return this.loginMethods[username];
+    }
+    const { data } = await this.client.get<
+      APIBasedAuth.LoginMethod[],
+      AxiosResponse<APIBasedAuth.LoginMethod[]>
+    >('/login-methods', { params: { login: username } });
+    this.loginMethods = { [username]: data };
+    return data;
+  }
```

Comment:
not putting any of this into storage as it seems pretty low overhead and we don't want to explode secure storage with tons of { username: loginMethods } values